### PR TITLE
Rholang: Initialize the registry for RholangCLI and the REPL

### PR DESCRIPTION
**Pull request by guardbotmk3**
## Overview
This PR makes the registry available to the REPL and RholangCLI. It also introduces a mechanism that should allow casper to initialize the registry at genesis.

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala [hunk 1]
```diff
@@ -41,6 +43,38 @@ class Runtime private (
     replaySpace.close()
     context.close()
   }
+  def injectEmptyRegistryRoot[F[_]]()(implicit F: Sync[F]): F[Unit] = {
```
Comment (birchmd) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: Empty parentheses are only needed to indicate a method with no arguments has side effects. I think you can drop the parentheses and wrap the `produce` calls in `Sync[F].delay` in order to leave this as a pure method.
```scala
for {
  spaceResult <- F.delay(space.produce(...))
  replayResult <- F.delay(replaySpace.produce(...))
  _ <- spaceResult match { ... }
} yield ()
```

Comment (goral09) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: Isn't this going to duplicate the work done in #1586 @birchmd ?

Comment (birchmd) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: I'll let your PR go though first and then update mine to call `injectEmptyRegistryRoot`

Comment (guardbotmk3) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: I've wrapped the side effects in F.

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala [hunk 2]
```diff
@@ -233,24 +276,25 @@ object Runtime {
...
+        ED25519_VERIFY                      -> SystemProcesses.ed25519Verify(space, dispatcher),
+        SHA256_HASH                         -> SystemProcesses.sha256Hash(space, dispatcher),
+        KECCAK256_HASH                      -> SystemProcesses.keccak256Hash(space, dispatcher),
+        BLAKE2B256_HASH                     -> SystemProcesses.blake2b256Hash(space, dispatcher),
+        SECP256K1_VERIFY                    -> SystemProcesses.secp256k1Verify(space, dispatcher),
+        REG_LOOKUP                          -> (registry.lookup(_)),
+        REG_LOOKUP_CALLBACK                 -> (registry.lookupCallback(_)),
+        REG_INSERT                          -> (registry.insert(_)),
+        REG_INSERT_CALLBACK                 -> (registry.insertCallback(_)),
+        REG_PUBLIC_REGISTER_INSERT_CALLBACK -> (registry.publicRegisterInsertCallback(_)),
```
Comment (goral09) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: Can you remind me what `publicRegisterInsertCallback` is doing?

**Pull request merged**